src/sentry/event_manager.py

```diff
@@ -688,6 +692,53 @@ def _pull_out_data(jobs, projects):
         )
 
 
+def _is_commit_sha(version: str):
+    return re.match(r"[0-9a-f]{40}", version) is not None
+
+
+def _associate_commits_with_release(release: Release, project: Project):
+    previous_release = release.get_previous_release(project)
+    possible_repos = (
+        RepositoryProjectPathConfig.objects.select_related(
+            "repository", "organization_integration", "organization_integration__integration"
+        )
+        .filter(project=project)
+        .all()
+    )
+    if possible_repos:
+        # If it does exist, kick off a task to look if the commit exists in the repository
+        target_repo = None
+        for repo_proj_path_model in possible_repos:
+            integration_installation = (
+                repo_proj_path_model.organization_integration.integration.get_installation(
+                    organization_id=project.organization.id
+                )
+            )
+            repo_client = integration_installation.get_client()
+            try:
+                repo_client.get_commit(
+                    repo=repo_proj_path_model.repository.name, sha=release.version
+                )
+                target_repo = repo_proj_path_model.repository
+                break
+            except ApiError as exc:
+                if exc.code != 404:
+                    raise
+
+        if target_repo is not None:
+            # If it does exist, fetch the commits for that repo
+            fetch_commits.apply_async(
```

Comment:
Does `fetch_commits` associated commits to a release? My apologies for not reading that function. It's a bit large.

As reference:
https://github.com/getsentry/sentry/blob/c80434508667a10b61c4edba1b962b0c1b3fe0e2/src/sentry/tasks/commits.py#L63-L231

Reply:
Yup, it handles model and relation creation. It also automatically fetches the last 10 commits if there is no previous release.